src/lib/factories.tsx

```diff
@@ -106,7 +107,13 @@ export function createShorthand(
 
   // Merge styles
   if (defaultProps.styles || overrideProps.styles || usersProps.styles) {
-    props.styles = _.merge(defaultProps.styles, usersProps.styles, overrideProps.styles)
+    props.styles = (...args) => {
+      return _.merge(
```

Comment:
don't we want to have `mergeStyles` function that would encapsulate this logic? Similar semantics is expressed in the `mergeThemes` file:

```js
// mergeComponentStyles()
...
return _.merge(callable(originalTarget)(styleParam), callable(originalSource)(styleParam))
```

maybe we could encapsulate and reuse this logic in the following way:

```ts
mergeStyles(target: ComponentSlotStyle, ...styles: ComponentSlotStyle[]) {
  ...
}
```

Reply:
Agreed, will encapsulate this logic.

Reply:
Introduced, please review again.